src/crawlers/browser_crawler.js

```diff
@@ -508,6 +508,11 @@ export default class BrowserCrawler extends BasicCrawler {
             const proxyInfo = this.proxyConfiguration.newProxyInfo(launchContextExtends.session && launchContextExtends.session.id);
             launchContext.proxyUrl = proxyInfo.url;
             launchContextExtends.proxyInfo = proxyInfo;
+
+            if (this.proxyConfiguration.hasRestrictedResidentialProxy) {
+                // @szmarczak: What's the difference with launchContextExtends?
```

Comment:
i guess that is just a (safe) way to extend the context, see https://github.com/apify/browser-pool/blob/ed1c6adddbd9d066ec0750db2db50dd6aed08362/src/launch-context.ts#L99